AwsResources.py

```diff
@@ -0,0 +1,198 @@
+                self.ec2_resource = boto3.resource('ec2', region_name=self.region_name)
+            else:
+                self.ec2_client = boto3.client('ec2')
+                self.ec2_resource = boto3.resource('ec2')
+        except Exception as e:
+            logging.error('Impossible logging in:')
+            logging.error(e)
+            logging.error(
+                'Please read https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#configuration '
+                'for a proper configuration of the credentials.')  # TODO: write a guide on our wiki
+
+            raise Exception('Critical error, aborting.')
+
+    def create_ebs(self):
+        tag_specifications = [{
+            'ResourceType': 'volume',
+            'Tags': [{
+                'Key': 'Name',
+                'Value': self.volume_name
+            }]
+        }]
+
+        # botocore complains if iops is none:
+        # Invalid type for parameter Iops, value: None, type: <class 'NoneType'>,
+        # valid types: <class 'int'>
+        try:
+            if self.iops is not None:
+                response = self.ec2_client.create_volume(
+                    AvailabilityZone=self.availability_zone,
+                    Encrypted=self.encrypted,
+                    Iops=self.iops,
+                    Size=self.size,
+                    VolumeType=self.volume_type,
+                    TagSpecifications=tag_specifications
+                )
+            else:
+                response = self.ec2_client.create_volume(
+                    AvailabilityZone=self.availability_zone,
+                    Encrypted=self.encrypted,
+                    Size=self.size,
+                    VolumeType=self.volume_type,
+                    TagSpecifications=tag_specifications
+                )
+        except Exception as e:
+            logging.error('Impossible creating the EBS volume:')
+            logging.error(e)
+            return -1
+
+        return response
+
+    def find_ami(self) -> str:
+        images = self.ec2_resource.images.filter(
+            Filters=[
+                {
+                    'Name': 'architecture',
+                    'Values': ['x86_64']
+                },
+                {
+                    'Name': 'name',
+                    'Values': ['ubuntu/images/hvm-ssd/ubuntu-xenial-*']
+                    # TODO: find a more appropriate AMI
+                }
+            ],
+            Owners=['099720109477']  # Canonical
+        )
+
+        sorted_list = sorted(images, reverse=True,
+                             key=lambda i: strptime(i.creation_date, '%Y-%m-%dT%H:%M:%S.%fZ'))
+
+        return [image for image in sorted_list][0].id
+
+    def create_key_pair(self):
+        response = self.ec2_client.create_key_pair(KeyName=self.key_name)
+        key_file = open(f'/tmp/{self.key_name}.pem', 'w')
+        key_file.write(response['KeyMaterial'])
```

Comment:
`key_file.close()` is missing. Alternatively, you can use the `with open(...) as f:` syntax.

Reply:
Good catch, thanks